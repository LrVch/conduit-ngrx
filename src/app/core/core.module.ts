import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
