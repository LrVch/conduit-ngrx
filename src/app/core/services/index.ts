export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
