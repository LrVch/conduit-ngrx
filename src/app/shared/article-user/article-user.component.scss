.article-user__avatar {
    display: flex;
    overflow: hidden;
    img {
        max-width: 100%;
    }
}

.article-user__button-container {
    display: flex;
}

a.article-user__author {
    color: #000;
    text-decoration: none;
    &:hover {
        opacity: .8;
    }
}

mat-card {
    padding-left: 0 !important;
    padding-top: 0 !important;
    box-shadow: none !important;
}