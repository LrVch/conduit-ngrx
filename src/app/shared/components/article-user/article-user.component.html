<mat-card>
  <mat-card-header>
    <div mat-card-avatar
         class="article-user__avatar">
      <a [routerLink]="['/profile', article.author.username]">
        <img appAvatar
             [src]="article.author.image"
             [attr.alt]="article.author.username">
      </a>
    </div>
    <mat-card-title>
      <a [routerLink]="['/profile', article.author.username]"
         class="article-user__author">{{ article.author.username }}</a>
    </mat-card-title>
    <mat-card-subtitle>{{article.createdAt | date: 'longDate' : '' : locale}}</mat-card-subtitle>
  </mat-card-header>
</mat-card>

<!-- canModify -->
<span [hidden]="!canModify">
  <a [routerLink]="['/editor', article.slug]"
     mat-stroked-button
     color="primary">Edit Article</a>
  &nbsp;
  <button (click)="onDeleteArticle()"
          mat-stroked-button
          [disabled]="isDeletingArticle"
          color="warn">Delete Article</button>

</span>

<!-- !canModify -->
<span [hidden]="canModify">
  <div class="article-user__button-container">
    <app-follow-button [isLoading]="article.author.isFollowLoading"
                       [followed]="article.author.following"
                       (followedToggle)="onToggleFollowing(article.author)">
      {{ article.author.username }}
    </app-follow-button>
    &nbsp;
    <button [disabled]="article.favoriting"
            mat-stroked-button
            color="primary">
      <app-favorite (favoritedToggle)="onFavoriveToggle()"
                    [isLoading]="article.favoriting"
                    [favorited]="article.favorited"
                    class="article__favorite">
        {{ article.favoritesCount }}
      </app-favorite>
    </button>
  </div>
</span>