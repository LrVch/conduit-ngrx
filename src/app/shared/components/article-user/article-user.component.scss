.article-user__avatar {
    display: flex;
    overflow: hidden;
    img {
        max-width: 100%;
    }
}

.article-user__button-container {
    display: flex;
    margin-top: 20px;
}

a.article-user__author {
    color: #000;
    text-decoration: none;
    &:hover {
        opacity: .8;
    }
}

.article-user__btn {
    margin-top: 20px;
}