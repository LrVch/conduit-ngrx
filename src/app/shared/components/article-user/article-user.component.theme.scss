@import '~@angular/material/theming';

@mixin conduit-article-user-component-theme($theme) {
  $primary: map-get($theme, primary);
  $accent: map-get($theme, accent);
  $foreground: map-get($theme, foreground);
  $background: map-get($theme, background);

  app-article-user {
    mat-card-title {
      .article-user__author {
        color: mat-color($foreground, text);
      }
    }
  }
}
