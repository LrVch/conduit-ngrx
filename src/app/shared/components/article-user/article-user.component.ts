import { Component, Input, Output, EventEmitter } from '@angular/core';
import { Article, Profile } from '@app/core';

@Component({
  selector: 'app-article-user',
  templateUrl: './article-user.component.html',
  styleUrls: ['./article-user.component.scss']
})
export class ArticleUserComponent {
  @Input('locale') locale = 'en';
  @Input('article') article: Article = {} as Article;
  @Input('canModify') canModify = false;
  @Input('isDeletingArticle') isDeletingArticle = false;
  @Output() favorited = new EventEmitter<Article>();
  @Output() folowing = new EventEmitter<Profile>();
  @Output() delete = new EventEmitter<Article>();

  onToggleFollowing(profile: Profile): void {
    this.folowing.emit(profile);
  }

  onFavoriveToggle(): void {
    this.favorited.emit(this.article);
  }

  onDeleteArticle(): void {
    this.delete.emit(this.article);
  }
}
