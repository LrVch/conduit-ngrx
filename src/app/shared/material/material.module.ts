import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
];

@NgModule({
  imports: modules,
  exports: modules
})
export class MaterialModule { }
