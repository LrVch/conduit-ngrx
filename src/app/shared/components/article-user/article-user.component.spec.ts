import { async, ComponentFixture } from '@angular/core/testing';
import { ConfigureFn, configureTests } from '@app/lib/testing';
import { ArticleUserComponent } from './article-user.component';
import {
  DebugElement,
  Component,
  Input,
  Output,
  EventEmitter
} from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '@app/shared/material/material.module';
import { RouterLinkDirectiveStubDirective } from '@app/lib/testing/directive/router-link-directive-stub';
import { Article, Profile } from '@app/core';
import { getArticle, getProfile } from '@app/lib/testing/mock-data.helpers';
import { TranslateModule } from '@ngx-translate/core';

@Component({
  selector: 'app-follow-button',
  template: `
    <a (click)="onFollowedToggle()"><ng-content></ng-content></a>
  `
})
class FollowComponent {
  @Input('isLoading') isLoading: boolean;
  @Input('followed') favorited: boolean;
  @Output() followedToggle = new EventEmitter<Article>();

  onFollowedToggle(): void {
    this.followedToggle.emit();
  }
  constructor() {}
}

@Component({
  selector: 'app-favorite',
  template: `
    <a (click)="onFavoritedToggle()"><ng-content></ng-content></a>
  `
})
class FavoriteComponent {
  @Input('isLoading') isLoading: boolean;
  @Input('favorited') favorited: boolean;
  @Output() favoritedToggle = new EventEmitter<Article>();

  onFavoritedToggle(): void {
    this.favoritedToggle.emit();
  }
  constructor() {}
}

describe('ArticleUserComponent', () => {
  let component: ArticleUserComponent;
  let fixture: ComponentFixture<ArticleUserComponent>;
  let de: DebugElement;
  let el: HTMLElement;
  const initialArticle: Article = {
    author: {},
    tagList: []
  } as Article;
  let linkDes;
  let routerLinks;
  let event;

  const article: Article = getArticle();

  beforeEach(async(() => {
    event = {
      preventDefault() {}
    };
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [
          ArticleUserComponent,
          RouterLinkDirectiveStubDirective,
          FavoriteComponent,
          FollowComponent
        ],
        imports: [MaterialModule, TranslateModule.forRoot()]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(ArticleUserComponent);
      component = fixture.componentInstance;
      component.article = initialArticle;
      de = fixture.debugElement;
      el = de.nativeElement;
      fixture.detectChanges();
      linkDes = de.queryAll(By.directive(RouterLinkDirectiveStubDirective));
      routerLinks = linkDes.map(deel =>
        deel.injector.get(RouterLinkDirectiveStubDirective)
      );
    });
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should has "article" @Input', () => {
    expect(component.article).toEqual(initialArticle);
  });

  it('should has "canModify" @Input', () => {
    expect(component.canModify).toEqual(false);
  });

  it('should has "isDeletingArticle" @Input', () => {
    expect(component.canModify).toEqual(false);
  });

  it('should show "article" @Input', () => {
    component.article = getArticle();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();

    component.article = getArticle();
    component.article.favoritesCount = 10;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should show "canModify" @Input', () => {
    component.article = getArticle();
    component.canModify = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should show "isDeletingArticle" @Input', () => {
    component.article = getArticle();
    component.isDeletingArticle = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should raise favorited event when clicked (favoritedToggle) and pass an selected article', () => {
    let selectedArticle;
    component.article = article;
    component.favorited.subscribe(a => (selectedArticle = a));

    fixture.detectChanges();

    const favoriteDe = de.query(By.css('app-favorite')).query(By.css('a'));

    favoriteDe.triggerEventHandler('click', event);
    expect(selectedArticle).toEqual(article);
  });

  it('should raise folowing event when clicked (followedToggle) and pass an selected profile', () => {
    let selectedProfile;
    component.article = article;

    component.folowing.subscribe(a => (selectedProfile = a));

    fixture.detectChanges();

    const followDe = de.query(By.css('app-follow-button')).query(By.css('a'));

    followDe.triggerEventHandler('click', event);
    expect(selectedProfile).toEqual(article.author);
  });

  it('should delete folowing event when clicked (onDeleteArticle) and pass an selected profile', () => {
    let deletedArticle;
    component.article = article;
    component.canModify = true;

    component.delete.subscribe(a => (deletedArticle = a));

    fixture.detectChanges();

    const deleteDe = de.query(By.css('[color="warn"]'));

    deleteDe.triggerEventHandler('click', event);

    expect(deletedArticle).toEqual(article);
  });

  it('can get RouterLinks from template', () => {
    component.article = article;
    fixture.detectChanges();

    expect(routerLinks.length).toBe(3, 'should have 3 routerLinks');
    expect(routerLinks[0].linkParams).toEqual([
      '/profile',
      article.author.username
    ]);
    expect(routerLinks[1].linkParams).toEqual([
      '/profile',
      article.author.username
    ]);
    expect(routerLinks[2].linkParams).toEqual(['/editor', article.slug]);
  });

  it('can click Articles link in template', () => {
    component.article = article;
    const goToProfileDe = linkDes[0];
    const goToProfileLink = routerLinks[0];

    expect(goToProfileLink.navigatedTo).toBeNull();

    fixture.detectChanges();
    goToProfileDe.triggerEventHandler('click', null);

    expect(goToProfileLink.navigatedTo).toEqual([
      '/profile',
      article.author.username
    ]);
  });
});
