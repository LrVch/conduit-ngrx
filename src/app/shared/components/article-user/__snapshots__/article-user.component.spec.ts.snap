// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleUserComponent should show "article" @Input 1`] = `
<app-article-user
  article={[Function Object]}
  canModify="false"
  delete={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  folowing={[Function EventEmitter]}
  isDeletingArticle="false"
  locale={[Function String]}
>
  <mat-card
    class="mat-card"
  >
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="article-user__avatar mat-card-avatar"
        mat-card-avatar=""
      >
        <a
          ng-reflect-link-params="/profile,username"
        >
          <img
            alt="username"
            appavatar=""
            src="image"
          />
        </a>
      </div>
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          <a
            class="article-user__author"
            ng-reflect-link-params="/profile,username"
          >
            username
          </a>
        </mat-card-title>
        <mat-card-subtitle
          class="mat-card-subtitle"
        >
          December 11, 2018
        </mat-card-subtitle>
      </div>
    </mat-card-header>
  </mat-card><span
    class=""
    hidden=""
  >
    <a
      aria-disabled="false"
      class="article-user__btn mat-stroked-button mat-primary"
      color="primary"
      mat-stroked-button=""
      ng-reflect-color="primary"
      ng-reflect-link-params="/editor,slug"
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        conduit.article-user.editButton
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
       
    <button
      class="article-user__btn mat-stroked-button mat-warn"
      color="warn"
      mat-stroked-button=""
      ng-reflect-color="warn"
      ng-reflect-disabled="false"
    >
      <span
        class="mat-button-wrapper"
      >
        conduit.article-user.deleteButton
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </span><span>
    <div
      class="article-user__button-container"
    >
      <app-follow-button
        ng-reflect-favorited="false"
        ng-reflect-is-loading="false"
      >
        <a>
           username 
        </a>
      </app-follow-button>
         
      <button
        class="mat-stroked-button mat-primary"
        color="primary"
        mat-stroked-button=""
        ng-reflect-color="primary"
        ng-reflect-disabled="false"
      >
        <span
          class="mat-button-wrapper"
        >
          <app-favorite
            class="article__favorite"
            ng-reflect-favorited="false"
            ng-reflect-is-loading="false"
          >
            <a>
               0 
            </a>
          </app-favorite>
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
    </div>
  </span>
</app-article-user>
`;

exports[`ArticleUserComponent should show "article" @Input 2`] = `
<app-article-user
  article={[Function Object]}
  canModify="false"
  delete={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  folowing={[Function EventEmitter]}
  isDeletingArticle="false"
  locale={[Function String]}
>
  <mat-card
    class="mat-card"
  >
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="article-user__avatar mat-card-avatar"
        mat-card-avatar=""
      >
        <a
          ng-reflect-link-params="/profile,username"
        >
          <img
            alt="username"
            appavatar=""
            src="image"
          />
        </a>
      </div>
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          <a
            class="article-user__author"
            ng-reflect-link-params="/profile,username"
          >
            username
          </a>
        </mat-card-title>
        <mat-card-subtitle
          class="mat-card-subtitle"
        >
          December 11, 2018
        </mat-card-subtitle>
      </div>
    </mat-card-header>
  </mat-card><span
    class=""
    hidden=""
  >
    <a
      aria-disabled="false"
      class="article-user__btn mat-stroked-button mat-primary"
      color="primary"
      mat-stroked-button=""
      ng-reflect-color="primary"
      ng-reflect-link-params="/editor,slug"
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        conduit.article-user.editButton
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
       
    <button
      class="article-user__btn mat-stroked-button mat-warn"
      color="warn"
      mat-stroked-button=""
      ng-reflect-color="warn"
      ng-reflect-disabled="false"
    >
      <span
        class="mat-button-wrapper"
      >
        conduit.article-user.deleteButton
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </span><span>
    <div
      class="article-user__button-container"
    >
      <app-follow-button
        ng-reflect-favorited="false"
        ng-reflect-is-loading="false"
      >
        <a>
           username 
        </a>
      </app-follow-button>
         
      <button
        class="mat-stroked-button mat-primary"
        color="primary"
        mat-stroked-button=""
        ng-reflect-color="primary"
        ng-reflect-disabled="false"
      >
        <span
          class="mat-button-wrapper"
        >
          <app-favorite
            class="article__favorite"
            ng-reflect-favorited="false"
            ng-reflect-is-loading="false"
          >
            <a>
               10 
            </a>
          </app-favorite>
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
    </div>
  </span>
</app-article-user>
`;

exports[`ArticleUserComponent should show "canModify" @Input 1`] = `
<app-article-user
  article={[Function Object]}
  canModify={[Function Boolean]}
  delete={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  folowing={[Function EventEmitter]}
  isDeletingArticle="false"
  locale={[Function String]}
>
  <mat-card
    class="mat-card"
  >
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="article-user__avatar mat-card-avatar"
        mat-card-avatar=""
      >
        <a
          ng-reflect-link-params="/profile,username"
        >
          <img
            alt="username"
            appavatar=""
            src="image"
          />
        </a>
      </div>
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          <a
            class="article-user__author"
            ng-reflect-link-params="/profile,username"
          >
            username
          </a>
        </mat-card-title>
        <mat-card-subtitle
          class="mat-card-subtitle"
        >
          December 11, 2018
        </mat-card-subtitle>
      </div>
    </mat-card-header>
  </mat-card><span
    class=""
  >
    <a
      aria-disabled="false"
      class="article-user__btn mat-stroked-button mat-primary"
      color="primary"
      mat-stroked-button=""
      ng-reflect-color="primary"
      ng-reflect-link-params="/editor,slug"
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        conduit.article-user.editButton
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
       
    <button
      class="article-user__btn mat-stroked-button mat-warn"
      color="warn"
      mat-stroked-button=""
      ng-reflect-color="warn"
      ng-reflect-disabled="false"
    >
      <span
        class="mat-button-wrapper"
      >
        conduit.article-user.deleteButton
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </span><span
    hidden=""
  >
    <div
      class="article-user__button-container"
    >
      <app-follow-button
        ng-reflect-favorited="false"
        ng-reflect-is-loading="false"
      >
        <a>
           username 
        </a>
      </app-follow-button>
         
      <button
        class="mat-stroked-button mat-primary"
        color="primary"
        mat-stroked-button=""
        ng-reflect-color="primary"
        ng-reflect-disabled="false"
      >
        <span
          class="mat-button-wrapper"
        >
          <app-favorite
            class="article__favorite"
            ng-reflect-favorited="false"
            ng-reflect-is-loading="false"
          >
            <a>
               0 
            </a>
          </app-favorite>
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
    </div>
  </span>
</app-article-user>
`;

exports[`ArticleUserComponent should show "isDeletingArticle" @Input 1`] = `
<app-article-user
  article={[Function Object]}
  canModify="false"
  delete={[Function EventEmitter]}
  favorited={[Function EventEmitter]}
  folowing={[Function EventEmitter]}
  isDeletingArticle={[Function Boolean]}
  locale={[Function String]}
>
  <mat-card
    class="mat-card"
  >
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="article-user__avatar mat-card-avatar"
        mat-card-avatar=""
      >
        <a
          ng-reflect-link-params="/profile,username"
        >
          <img
            alt="username"
            appavatar=""
            src="image"
          />
        </a>
      </div>
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          <a
            class="article-user__author"
            ng-reflect-link-params="/profile,username"
          >
            username
          </a>
        </mat-card-title>
        <mat-card-subtitle
          class="mat-card-subtitle"
        >
          December 11, 2018
        </mat-card-subtitle>
      </div>
    </mat-card-header>
  </mat-card><span
    class=""
    hidden=""
  >
    <a
      aria-disabled="false"
      class="article-user__btn mat-stroked-button mat-primary"
      color="primary"
      mat-stroked-button=""
      ng-reflect-color="primary"
      ng-reflect-link-params="/editor,slug"
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        conduit.article-user.editButton
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
       
    <button
      class="article-user__btn mat-stroked-button mat-warn"
      color="warn"
      disabled=""
      mat-stroked-button=""
      ng-reflect-color="warn"
      ng-reflect-disabled="true"
    >
      <span
        class="mat-button-wrapper"
      >
        conduit.article-user.deleteButton
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="true"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </span><span>
    <div
      class="article-user__button-container"
    >
      <app-follow-button
        ng-reflect-favorited="false"
        ng-reflect-is-loading="false"
      >
        <a>
           username 
        </a>
      </app-follow-button>
         
      <button
        class="mat-stroked-button mat-primary"
        color="primary"
        mat-stroked-button=""
        ng-reflect-color="primary"
        ng-reflect-disabled="false"
      >
        <span
          class="mat-button-wrapper"
        >
          <app-favorite
            class="article__favorite"
            ng-reflect-favorited="false"
            ng-reflect-is-loading="false"
          >
            <a>
               0 
            </a>
          </app-favorite>
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
    </div>
  </span>
</app-article-user>
`;
